@lbailly 

```
recurring.go:21:6: exported type RecurringDetail should have comment or be unexported
recurring_gateway.go:11:1: comment on exported method RecurringGateway.ListRecurringDetails should be of the form "ListRecurringDetails ..."
```

2 lint errors to fix.

I've fixed unit tests on TravisCI for pull requests from forks, so now it's more useful

--- a/adyen.go
+++ b/adyen.go
@@ -78,7 +78,7 @@ func (a *Adyen) ClientURL() string {
 
 // adyenURL returns Adyen backend URL
 func (a *Adyen) adyenURL(service string, requestType string) string {
-	return a.Credentials.Env.BaseURL(service, APIVersion) + "/" + requestType
+	return a.Credentials.Env.BaseURL(service, APIVersion) + "/" + requestType + "/" + service + "/"
 }
 
 // createHPPUrl returns Adyen HPP url
@@ -103,13 +103,13 @@ func (a *Adyen) SetCurrency(currency string) {
 }
 
 // execute request on Adyen side, transforms "requestEntity" into JSON representation
-func (a *Adyen) execute(method string, service string, requestEntity interface{}) (*Response, error) {
+func (a *Adyen) execute(service string, method string, requestEntity interface{}) (*Response, error) {
 	body, err := json.Marshal(requestEntity)
 	if err != nil {
 		return nil, err
 	}
 
-	url := a.adyenURL(service, method) + "/" + service + "/"
+	url := a.adyenURL(service, method)
 	req, err := http.NewRequest("POST", url, bytes.NewBuffer(body))
 	if err != nil {
 		return nil, err

--- a/modification_gateway.go
+++ b/modification_gateway.go
@@ -17,7 +17,7 @@ type ModificationGateway struct {
 
 // Capture - Perform capture payment in Adyen
 func (a *ModificationGateway) Capture(req *Capture) (*CaptureResponse, error) {
-	resp, err := a.execute(captureType, PaymentService, req)
+	resp, err := a.execute(PaymentService, captureType, req)
 
 	if err != nil {
 		return nil, err
@@ -28,7 +28,7 @@ func (a *ModificationGateway) Capture(req *Capture) (*CaptureResponse, error) {
 
 // Cancel - Perform cancellation of the authorised transaction
 func (a *ModificationGateway) Cancel(req *Cancel) (*CancelResponse, error) {
-	resp, err := a.execute(cancelType, PaymentService, req)
+	resp, err := a.execute(PaymentService, cancelType, req)
 
 	if err != nil {
 		return nil, err
@@ -40,7 +40,7 @@ func (a *ModificationGateway) Cancel(req *Cancel) (*CancelResponse, error) {
 // CancelOrRefund - Perform cancellation for not captured transaction
 // otherwise perform refund action
 func (a *ModificationGateway) CancelOrRefund(req *Cancel) (*CancelOrRefundResponse, error) {
-	resp, err := a.execute(cancelOrRefundType, PaymentService, req)
+	resp, err := a.execute(PaymentService, cancelOrRefundType, req)
 
 	if err != nil {
 		return nil, err
@@ -51,7 +51,7 @@ func (a *ModificationGateway) CancelOrRefund(req *Cancel) (*CancelOrRefundRespon
 
 // Refund - perform refund for already captured request
 func (a *ModificationGateway) Refund(req *Refund) (*RefundResponse, error) {
-	resp, err := a.execute(refundType, PaymentService, req)
+	resp, err := a.execute(PaymentService, refundType, req)
 
 	if err != nil {
 		return nil, err

--- a/payment3d_gateway.go
+++ b/payment3d_gateway.go
@@ -10,7 +10,7 @@ const authorise3DType = "authorise3d"
 
 // Authorise3D - Perform authorise payment in Adyen
 func (a *PaymentGateway) Authorise3D(req *Authorise3D) (*AuthoriseResponse, error) {
-	resp, err := a.execute(authorise3DType, PaymentService, req)
+	resp, err := a.execute(PaymentService, authorise3DType, req)
 
 	if err != nil {
 		return nil, err

--- a/payment_gateway.go
+++ b/payment_gateway.go
@@ -13,7 +13,7 @@ const directoryLookupURL = "directory/v2"
 
 // AuthoriseEncrypted - Perform authorise payment in Adyen
 func (a *PaymentGateway) AuthoriseEncrypted(req *AuthoriseEncrypted) (*AuthoriseResponse, error) {
-	resp, err := a.execute(authoriseType, PaymentService, req)
+	resp, err := a.execute(PaymentService, authoriseType, req)
 
 	if err != nil {
 		return nil, err
@@ -24,7 +24,7 @@ func (a *PaymentGateway) AuthoriseEncrypted(req *AuthoriseEncrypted) (*Authorise
 
 // Authorise - Perform authorise payment in Adyen
 func (a *PaymentGateway) Authorise(req *Authorise) (*AuthoriseResponse, error) {
-	resp, err := a.execute(authoriseType, PaymentService, req)
+	resp, err := a.execute(PaymentService, authoriseType, req)
 
 	if err != nil {
 		return nil, err

--- a/recurring.go
+++ b/recurring.go
@@ -18,6 +18,7 @@ type RecurringDetailsResult struct {
 	ShopperReference         string `json:"shopperReference"`
 }
 
+// RecurringDetail structure to hold information associated to a recurring payment
 type RecurringDetail struct {
 	Acquirer        string `json:"acquirer"`
 	AcquirerAccount string `json:"acquirerAccount"`

--- a/recurring_gateway.go
+++ b/recurring_gateway.go
@@ -10,7 +10,7 @@ const listRecurringDetailsType = "listRecurringDetails"
 
 // ListRecurringDetails - Get list of recurring payments in Adyen
 func (a *RecurringGateway) ListRecurringDetails(req *RecurringDetailsRequest) (*RecurringDetailsResult, error) {
-	resp, err := a.execute(listRecurringDetailsType, RecurringService, req)
+	resp, err := a.execute(RecurringService, listRecurringDetailsType, req)
 
 	if err != nil {
 		return nil, err

--- a/recurring_gateway.go
+++ b/recurring_gateway.go
@@ -8,7 +8,7 @@ type RecurringGateway struct {
 // listRecurringDetailsType - listRecurringDetails type request, @TODO: move to enums
 const listRecurringDetailsType = "listRecurringDetails"
 
-// AuthoriseEncrypted - Perform authorise payment in Adyen
+// ListRecurringDetails - Get list of recurring payments in Adyen
 func (a *RecurringGateway) ListRecurringDetails(req *RecurringDetailsRequest) (*RecurringDetailsResult, error) {
 	resp, err := a.execute(listRecurringDetailsType, RecurringService, req)
 

